internal/controller/exporter_controller.go

```diff
@@ -70,6 +70,32 @@ func (r *ExporterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, err
 	}
 
+	// TODO: use field selector once KEP-4358 is stabilized
+	// Reference: https://github.com/kubernetes/kubernetes/pull/122717
+	var leases jumpstarterdevv1alpha1.LeaseList
+	err = r.List(
+		ctx,
+		&leases,
```

Comment:
We should filter by non-ended leases, otherwise this will get slower as we accumulate ended leases on the system.

Reply:
Ahh I see about the KEP, I propose we workaround for now by adding a label to the ended leases that we can filter with., Or an annotation.

Reply:
> We should filter by non-ended leases, otherwise this will get slower as we accumulate ended leases on the system.

That's also part of the reason why we want to expire/archive the leases.

> Ahh I see about the KEP, I propose we workaround for now by adding a label to the ended leases that we can filter with., Or an annotation.

The KEP is already implemented in k8s 1.30, I will try if it works.

Reply:
But openshift 4.16 (current stable is 1.29)

oc version
Client Version: 4.13.15
Kustomize Version: v4.5.7
Server Version: 4.16.11
Kubernetes Version: v1.29.7+d77deb8

Reply:
Implemented label based filtering in https://github.com/jumpstarter-dev/jumpstarter-controller/pull/37/commits/12df631e6b67c244bbb0c82a2e49aff3d8e01ad5

Reply:
Thank you, it looks great and will help for the time being.